xbmc/music/MusicDatabase.cpp

```diff
@@ -5719,6 +5964,23 @@ bool CMusicDatabase::GetFilter(CDbUrl &musicUrl, Filter &filter, SortDescription
   const CUrlOptions::UrlOptions& options = musicUrl.GetOptions();
   CUrlOptions::UrlOptions::const_iterator option;
 
+  //Process role options, common to artist and album type filtering
+  int idRole = -1;
+  option = options.find("roleid");
+  if (option != options.end())
+    idRole = (int)option->second.asInteger();
+  else
+  {
+    option = options.find("role");
+    if (option != options.end())
+      idRole = GetRoleByName(option->second.asString());
+  }
+  if (idRole == -1)
+    idRole = 1;  //Restrict to song_artist to "artists" only, no other roles.
+  
+  std::string strRoleSQL; //Role < -1 means all roles, otherwise filter by role
+  if(idRole > 0) strRoleSQL = PrepareSQL(" AND song_artist.idRole = %i ", idRole);  
```

Comment:
can idRole ever be 0? wondering because it's either 1 or something larger I assume reading the flow.